### ISSUE TITLE
Encoding error

### ISSUE THREAD

**Issue by rabenklau** [OpenPoGo/OpenPoGoBot#134]
When running latest master, im geting thrown ascii/unicode
Errors. Same as have been documented in pokemongoBot. 

I did not find a bugreport here yet, so... Thought i just file it as good as im currently able.
### Steps to Reproduce

Visit pokestop with unicode characters in name.
### Other Information

OS: win10, archlinux

**Comment by wchill** [comment]
Please provide stack trace, as this should have been fixed already.

**Comment by fonix232** [comment]
Stack trace:

EDIT:
Thanks to Github not preserving code formatting, here you go: http://pastebin.com/Ys7ivZdX

**Comment by wchill** [comment]
@rabenklau @fonix232 please try https://github.com/Drekin/win-unicode-console and see if this helps.

**Comment by Sjolus** [comment]
But what if I'm on Ubuntu?

```
[2016-08-01 07:45:13] [#] Moving closer to Grumpy
[2016-08-01 07:45:13] [#] Walking from (59.333040985662805, 18.097211083672967) to (59.333054, 18.097168) for approx. 2.00 seconds
[2016-08-01 07:45:13] [#] Walking Finished
[2016-08-01 07:45:15] [#] Now at Pokestop: Grumpy
[2016-08-01 07:45:15] [PokeStop] Spinning...
[2016-08-01 07:45:19] Web client disconnected
[2016-08-01 07:45:19] [PokeStop] Loot:
[2016-08-01 07:45:19] [PokeStop] + 50 XP
[2016-08-01 07:45:19] [PokeStop] + 3 Pokeballs
[2016-08-01 07:45:19] [PokeStop] PokeStop is on cooldown for 4.99 minutes.
Traceback (most recent call last):
  File "./pokecli.py", line 293, in <module>
    main()
  File "./pokecli.py", line 286, in main
    bot.take_step()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 76, in take_step
    self.stepper.take_step()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 41, in take_step
    self._work_at_position(position[0], position[1])
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 62, in work
    self.stepper.walk_to(self.config.walk, *position)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 68, in walk_to
    self._do_walk_to(speed, position_lat, position_lng, lat, lng, alt, 25)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 94, in _do_walk_to
    self._work_at_position(position_lat, position_lng)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/stepper.py", line 137, in _work_at_position
    self.bot.work_on_cell(cell)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/__init__.py", line 90, in work_on_cell
    walk_worker.work()
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/cell_workers/walk_towards_fort_worker.py", line 55, in work
    logger.log(u"[#] Found fort {} at distance {}".format(fort_name, format_dist(dist, unit)))
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/logger.py", line 26, in log
    manager.fire("logging", text=string, color=color, prefix=prefix)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/event_manager.py", line 101, in fire
    return self.events[event_name].fire(*args, **kwargs)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/event_manager.py", line 64, in fire
    return_dict = listener(**listener_args)
  File "/home/sjolus/OpenPoGoBot/pokemongo_bot/logger.py", line 46, in _log
    print(output)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 43: ordinal not in range(128)
```

**Comment by fonix232** [comment]
@wchill thanks, it helps, though it is a nuisance to install another package and use that. It could be added with an if platform switch.

**Comment by wchill** [comment]
Yes, now that I know that that package helps on Windows we can figure something out.

**Comment by ahmet2mir** [comment]
Same issue with a french char `à` (\xe0) and all other french char I think

```
File "/apps/OpenPoGoBot/pokemongo_bot/logger.py", line 46, in _log
    print(output)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe0' in position 45: ordinal not in range(128)
```

**Comment by wchill** [comment]
should be closed by #156, please reopen if there are still issues.

**Comment by Sjolus** [comment]
@wchill Yup, solved most issues - except it was now printing stuff like

`b'[2016-08-01 14:45:12] [#] Walking Finished'`

At least it's not crashing now, which is perfect. To make it print prettier I ugly-hacked it with some Stack Overflow googling so the print-row now is:

``` python
print("".join(map(chr, output.encode('utf-8'))))
```

Works perfect for me (since the terminal reading it is utf-8 itself) and logs look like this:

```
[2016-08-01 14:50:20] [+] No more free incubators
```

**Comment by fonix232** [comment]
Did not solve all issues for me.

Trace: http://pastebin.com/TuWMLChf

**Comment by ahmet2mir** [comment]
Not fixed with commit 6e22239f214e825d668124aacd570920077b03eb
with --location "Lille" (france)

```
Traceback (most recent call last):
  File "pokecli.py", line 316, in <module>
    main()
  File "pokecli.py", line 309, in main
    bot.run()
  File "/apps/openpogobot/pokemongo_bot/__init__.py", line 92, in run
    self.navigator.navigate(map_cells)
  File "/apps/openpogobot/pokemongo_bot/navigation/fort_navigator.py", line 49, in navigate
    logger.log("[#] Now at Pokestop: " + fort_name)
  File "/apps/openpogobot/pokemongo_bot/logger.py", line 26, in log
    manager.fire("logging", text=string, color=color, prefix=prefix)
  File "/apps/openpogobot/pokemongo_bot/event_manager.py", line 101, in fire
    return self.events[event_name].fire(*args, **kwargs)
  File "/apps/openpogobot/pokemongo_bot/event_manager.py", line 64, in fire
    return_dict = listener(**listener_args)
  File "/apps/openpogobot/pokemongo_bot/logger.py", line 47, in _log
    print(output.decode('utf-8'))
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "/usr/lib/python2.7/site-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 51: ordinal not in range(128)
```

**Comment by khllkcm** [comment]
Ran into this in New York.

```
Traceback (most recent call last):

  File "pokecli.py", line 347, in <module>

    main()

  File "pokecli.py", line 340, in main

    bot.run()

  File "/bots/OpenPoGoBot/pokemongo_bot/__init__.py", line 91, in run

    self.work_on_cells(map_cells)

  File "/bots/OpenPoGoBot/pokemongo_bot/__init__.py", line 98, in work_on_cells
    self.fire("pokestops_found", pokestops=cell.pokestops)

  File "/bots/OpenPoGoBot/pokemongo_bot/__init__.py", line 85, in fire

    manager.fire_with_context(event, self, *args, **kwargs)

  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 107, in fire_with_context

    return self.fire(event_name, *args, **kwargs)

  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 101, in fire

    return self.events[event_name].fire(*args, **kwargs)

  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 64, in fire

    return_dict = listener(**listener_args)

  File "./plugins/spin_pokestop/__init__.py", line 45, in visit_near_pokestops

    manager.fire_with_context('pokestop_arrived', bot, pokestop=pokestop)
  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 107, in fire_with_context

    return self.fire(event_name, *args, **kwargs)

  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 101, in fire

    return self.events[event_name].fire(*args, **kwargs)

  File "/bots/OpenPoGoBot/pokemongo_bot/event_manager.py", line 64, in fire

    return_dict = listener(**listener_args)

  File "./plugins/spin_pokestop/__init__.py", line 72, in spin_pokestop

    format_dist(dist, bot.config.distance_unit)), color="yellow")

UnicodeEncodeError: 'ascii' codec can't encode character u'\xe1' in position 3: ordinal not in 
range(128)
```

Removing "fort_details["fort"].fort_name" from line 71 in plugins/spin_pokestop/**init**.py fixed it for me.

I also removed another occurrence of fort_name in /pokemongo_bot/navigation/fort_navigator.py.
